Bonsai.Tinkerforge/UidConverter.cs

```diff
@@ -0,0 +1,111 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.ComponentModel;
+using System.Globalization;
+using Bonsai.Expressions;
+using System.Threading;
+using Tinkerforge;
+
+namespace Bonsai.Tinkerforge
+{
+    internal class UidConverter : StringConverter
+    {
+        // Cache for discovered Brick / Bricklet devices
+        public Dictionary<string, DeviceData> devices;
```

Comment:
Can we move the `DeviceData` class definition inside the `UidConverter` class? It doesn't seem to be used anywhere else so it would make it more explicit that it is an implementation detail.

Also, do we need this member to be `public`?